### PR TITLE
[Kernel] changing fused moe kernel chunk size default to 32k

### PR DESCRIPTION
**Pull request by avshalomman**
This PR changes the default value of VLLM_FUSED_MOE_CHUNK_SIZE from 64k to 32k, which lowers memory consumption during the forward pass.
We've measured the effects on kernel latency and found them to be minimal (see table below).
These numbers are for A100 80GB, we also tested on an H100 and found similar results.

```
|   intermediate_size |   32k chunk |   64k chunk | slowdown (%)   |
|--------------------:|------------:|------------:|:---------------|
|                1536 |     140.117 |     139.287 | 0.6%           |
|                3072 |     271.47  |     270.644 | 0.3%           |
|                6144 |     617.339 |     615.37  | 0.3%           |
|               12288 |    1442     |    1438.64  | 0.2%           |
```

For clarity, here is how we called the kernel to measure perf (we warmed up the kernel + averaged multiple iterations ofc).

```
fused_moe(a,  # shape = [256K, 4K]
          w1, # shape = [16, 2 * intermediate_size, 4K]
          w2, # shape = [16, intermediate_size, 4K]
          router_logits, # shape = [256K, 16]
          topk, # = 2
          inplace = True,
          renormalize = False)
```

### CODE REVIEW

**Review by mgoin — APPROVED** [review]
This seems reasonable to me but @comaniac should sign off before landing. Can you comment on the memory usage scenario that was the issue?

**Comment by comaniac** [comment]
I'm ok with this change. We set to 64k because the kernel is likely having illegal memory access issue with >64k tokens.

**Review by comaniac — APPROVED** [review]

**Comment by gshtras** [comment]
With this change we observe a regression (GPU memory access error) on Mixtral-8x22B-v0.1 with dynamic fp8; batch size=2000 input-length=2048:
`benchmarks/benchmark_throughput.py --model mistralai/Mixtral-8x22B-Instruct-v0.1 --input-len 2048 --output-len 128 --num-prompts 2000 --tensor-parallel-size 4 --quantization fp8`

Verified on both H100 and MI300

@avshalomman @comaniac FYI

**Pull request merged**
